### PR TITLE
Add logging to Forge wrapper and auto-detect cloud

### PR DESCRIPTION
**Pull request by sionescu**
### Description

Add logging for all output of the Forge wrapper.
Auto-detect cloud environment from the cluster name.

### Test Plan

Successfully scheduled "land_blocking" tests on the GCP clusters.

### CODE REVIEW

**Review by rustielin — APPROVED** [review]
LGTM! Thanks for improving the code quality with the logging

Stamping to unblock, but pls fix the unit tests too: https://github.com/aptos-labs/aptos-core/actions/runs/4640550086/jobs/8212607391?pr=7646

File: testsuite/forge.py [hunk 1]
```diff
@@ -37,6 +38,17 @@
 from forge_wrapper_core.time import SystemTime, Time
 from forge_wrapper_core.cluster import Cloud, ForgeCluster, ForgeJob
 
+log = logging.getLogger("")
```
Comment (perryjrandall) [inline testsuite/forge.py:41]: put this in a function and call that from either main or a decorator on commands

File: testsuite/forge.py [hunk 2]
```diff
@@ -1227,13 +1241,17 @@ def test(
...
 
+    if verbose:
+        log.setLevel(logging.DEBUG)
+
     ### XXX: hack these arguments to force Forge to run with overrides
-    # cloud = "gcp"
     # forge_cluster_name = "aptos-forge-0"
     # forge_enable_performance = "true"
 
+    log.debug("Before init")
```
Comment (perryjrandall) [inline testsuite/forge.py:1251]: remove

File: testsuite/forge.py [hunk 3]
```diff
@@ -1227,13 +1241,17 @@ def test(
...
+
     # Initialize all configs
-    shell = LocalShell(verbose == "true")
+    shell = LocalShell(verbose=verbose)
```
Comment (perryjrandall) [inline testsuite/forge.py:1254]: nice!

File: testsuite/forge.py [hunk 4]
```diff
@@ -1242,10 +1260,10 @@ def test(
     config = ForgeConfig(S3ForgeConfigBackend(context, DEFAULT_CONFIG))
     config.init()
 
+    log.debug("After init")
```
Comment (perryjrandall) [inline testsuite/forge.py:1263]: remove

File: testsuite/forge.py [hunk 5]
```diff
@@ -1311,14 +1329,16 @@ def test(
     assert forge_cluster_name, "Forge cluster name is required"
 
     # cloud
-    if cloud.upper() == "AWS":
+    if cloud is not None:
```
Comment (perryjrandall) [inline testsuite/forge.py:1332]: how can we tell if this feature is in use? (I think we need some kind of logging each invocation in humio!)

File: testsuite/forge.py [hunk 6]
```diff
@@ -1311,14 +1329,16 @@ def test(
...
+        log.warning(
+            "Explicitly setting the cloud is deprecated. The cloud is now inferred from the cluster name."
+        )
+    if "big" in forge_cluster_name:
```
Comment (perryjrandall) [inline testsuite/forge.py:1336]: hehe

**Review by perryjrandall — APPROVED** [review]
love it!

File: testsuite/forge.py (hunk 5 above)
Comment (sionescu) [reply testsuite/forge.py:1349]: We'll need to come up with a more comprehensive solution of how to log to Humio from Forge.

File: testsuite/forge.py (hunk 6 above)
Comment (sionescu) [reply testsuite/forge.py:1353]: Yeah :D

**Pull request merged**